Clean up changelog in preparation for Rojo 7.4.0 (#788)
rojo-rbx/rojo: CHANGELOG.md

# Rojo Changelog

## Unreleased Changes
### Additions
* Added support for `.toml` files to `$path` ([#633])
* Significantly improved performance of `rojo sourcemap` ([#668])
* Add `plugin` flag to the `build` command that outputs to the local plugins folder ([#735])
* Added `Terrain` classname inference, similar to services ([#771])

	`Terrain` may now be defined in the projects without using `$classname`:
	```json
	"Workspace": {
		"Terrain": {
			"$path": "path/to/terrain.rbxm"
		}
	}
	```

* Added support for `Terrain.MaterialColors` ([#770])

	`Terrain.MaterialColors` is now represented in projects in a human readable format:
	```json
	"Workspace": {
		"Terrain": {
			"$path": "path/to/terrain.rbxm"
			"$properties": {
				"MaterialColors": {
					"Grass": [10, 20, 30],
					"Asphalt": [40, 50, 60],
					"LeafyGrass": [255, 155, 55]
				}
			}
		}
	}
	```

> * Added better support for `Font` properties ([#731])

Reviewer: Should this one include an example?

* Add Confirmation Behavior setting ([#774])
* Select Instances from diff tree view ([#709])
* Added sync reminder notification. ([#689])
* Added protection against syncing a model to a place. ([#691])
* Add new plugin template to the `init` command ([#738])
* Added rich Source diffs in patch visualizer ([#748])

> * Patch visualizer now indicates what changes failed to apply. ([#717])

Reviewer: Might be nice to include the image for this one
![image](https://github.com/rojo-rbx/rojo/assets/40185666/f08399ef-1e16-4f1c-bed8-552821f98cef)

* Add buttons for navigation on the Connected page ([#722])
* Improved sync info text on Connected page. ([#692])

### Fixes
* Fixed the diff visualizer of connected sessions. ([#674])
* Fixed disconnected session activity. ([#675])
* Skip confirming patches that contain only a datamodel name change. ([#688])
* Fix Rojo breaking when users undo/redo in Studio ([#708])
* Improve tooltip behavior ([#723])
* Better settings controls ([#725])
* Rework patch visualizer with many fixes and improvements ([#713], [#726], [#755])

> * Don't override the initial enabled state for source diffing ([#760])

Reviewer: This is an internal detail regarding Roact's weird interaction with plugin guis and doesn't need to be in the log at all imo

* Added rich Source diffs in patch visualizer ([#748])

	A "View Diff" button for script sources is now present in the patch visualizer. Clicking it displays a side-by-side diff of the script changes:

	![The patch visualizer contains a "view diff" button](https://user-images.githubusercontent.com/40185666/256065992-3f03558f-84b0-45a1-80eb-901f348cf067.png)

	![The "View Diff" button opens a widget that displays a diff](https://user-images.githubusercontent.com/40185666/256066084-1d9d8fe8-7dad-4ee7-a542-b4aee35a5644.png)

* Patch visualizer now indicates what changes failed to apply. ([#717])

	A clickable warning label is displayed when the Rojo plugin is unable to apply changes. Clicking the label displays precise information about which changes failed:

	![Patch visualizer displays a clickable warning label when changes fail to apply](https://user-images.githubusercontent.com/40185666/252063660-f08399ef-1e16-4f1c-bed8-552821f98cef.png)

#### Miscellaneous
* Added `plugin` flag to the `build` command that outputs to the local plugins folder ([#735])

> 	This is a flag that builds a Rojo project into Roblox Studio's plugins directory. This allows you to build a Rojo project and load it into Studio as a plugin without having to type the full path to the plugins directory. It can be used like this: `rojo build <PATH-TO_PROJECT> --plugin <FILE-NAME>`

Reviewer 2: ```suggestion
	This is a flag that builds a Rojo project into Roblox Studio's plugins directory. This allows you to build a Rojo project and load it into Studio as a plugin without having to type the full path to the plugins directory. It can be used like this: `rojo build <PATH-TO-PROJECT> --plugin <FILE-NAME>`
```

Author: Heh, nice catch

# Rojo Changelog

## Unreleased Changes
### Additions
* Added support for `.toml` files to `$path` ([#633])
* Significantly improved performance of `rojo sourcemap` ([#668])

> * Add `plugin` flag to the `build` command that outputs to the local plugins folder ([#735])

Reviewer 3: ```suggestion
* Added a `plugin` flag to the `build` command that outputs to the local plugins folder ([#735])
```

* Added `Terrain` classname inference, similar to services ([#771])

	`Terrain` may now be defined in the projects without using `$classname`:
	```json
	"Workspace": {
		"Terrain": {
			"$path": "path/to/terrain.rbxm"
		}
	}
	```

* Added support for `Terrain.MaterialColors` ([#770])

	`Terrain.MaterialColors` is now represented in projects in a human readable format:
	```json
	"Workspace": {
		"Terrain": {
			"$path": "path/to/terrain.rbxm"
			"$properties": {
				"MaterialColors": {
					"Grass": [10, 20, 30],
					"Asphalt": [40, 50, 60],
					"LeafyGrass": [255, 155, 55]
				}
			}
		}
	}
	```

* Added better support for `Font` properties ([#731])

	`FontFace` properties may now be defined using implicit property syntax:
	```json
	"TextBox": {
		"$className": "TextBox",
		"$properties": {
			"FontFace": {
				"family": "rbxasset://fonts/families/RobotoMono.json",
				"weight": "Thin",
				"style": "Normal"
			}
		}
	}
	```

> * Add Confirmation Behavior setting ([#774])

Reviewer 3: This feels like it will cut off questions about this particular item
```suggestion
* Addded a setting to control patch confirmation behavior ([#774])
```

Author: I went ahead and added an explanation too

> * Select Instances from diff tree view ([#709])

Reviewer 3: ```suggestion
* Added the ability to select Instances in patch visualizer ([#709])
```

> * Added sync reminder notification. ([#689])

Reviewer 3: Could do with added clarity on this:
```suggestion
* Added a reminder notification to start syncing to a place you previously synced to. ([#689])
```

* Added protection against syncing a model to a place. ([#691])

> * Add new plugin template to the `init` command ([#738])

Reviewer 3: ```suggestion
* Added new plugin template to the `init` command ([#738])
```
